Tasks/src/provider/Consumer.java

```diff
@@ -0,0 +1,14 @@
+package provider;
+
+import java.util.List;
+
+class Consumer {
+    public static void consume(final List<? extends A> as) {
```

Comment:
Although `B` is a subtype of `A,` `List<B>` is not a subtype of `List<A>` and, in fact, these two types are not related. The common parent of `List<A>` and `List<B>` is `List<?>.`

In order to create a relationship between these classes so that the code can access `A`'s methods through `List<B>`'s elements, we need to use an upper bounded wildcard.